# Name: What Makes a Google Play App Successful?

Goal: Explore how category, type (free or paid), price, size, and last
update date relate to app installs and ratings, to help developers
decide what kind of new app to build.

## Asking Questions
Q1: Which category has the highest average rating?

Q2: Do free apps have more installs than paid apps on average?

Q3: Do apps updated more recently have higher average ratings?

Q4: Which category has the highest percentage of positive reviews?

Q5: Is there a relationship between app size and number of installs?

## Data Wrangling
### Gather


First, we import the libraries we will use throughout the project: NumPy, Pandas, Matplotlib and Seaborn.

In [43]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt



Then, we load the apps and reviews CSV files using relative paths, so the notebook runs on any computer.

In [44]:
df_apps_raw = pd.read_csv('data/raw/googleplaystore.csv')
df_reviews_raw = pd.read_csv('data/raw/googleplaystore_user_reviews.csv')


### Assess

In [45]:
df_apps_raw.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [46]:
df_reviews_raw.head()

,App,Translated_Review,Sentiment,Sentiment_Polarity,Sentiment_Subjectivity
0,10 Best Foods for You,I like eat delicious food. That's I'm cooking ...,Positive,1.00,0.533333
1,10 Best Foods for You,This help eating healthy exercise regular basis,Positive,0.25,0.288462
2,10 Best Foods for You,NaN,NaN,NaN,NaN
3,10 Best Foods for You,Works great especially going grocery store,Positive,0.40,0.875000
4,10 Best Foods for You,Best idea us,Positive,1.00,0.300000


In [47]:
print('Apps table shape:')
print(df_apps_raw.shape)

Apps table shape:
(10841, 13)


In [48]:
print('Reviews table shape:')
print(df_reviews_raw.shape)

Reviews table shape:
(64295, 5)


In [49]:
df_apps_raw.info()

<class 'pandas.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  str    
 1   Category        10841 non-null  str    
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  str    
 4   Size            10841 non-null  str    
 5   Installs        10841 non-null  str    
 6   Type            10840 non-null  str    
 7   Price           10841 non-null  str    
 8   Content Rating  10840 non-null  str    
 9   Genres          10841 non-null  str    
 10  Last Updated    10841 non-null  str    
 11  Current Ver     10833 non-null  str    
 12  Android Ver     10838 non-null  str    
dtypes: float64(1), str(12)
memory usage: 1.1 MB


In [50]:
df_reviews_raw.info()

<class 'pandas.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  str    
 1   Translated_Review       37427 non-null  str    
 2   Sentiment               37432 non-null  str    
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), str(3)
memory usage: 2.5 MB


In [51]:
df_apps_raw.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [52]:
df_reviews_raw.isnull().sum()

App                           0
Translated_Review         26868
Sentiment                 26863
Sentiment_Polarity        26863
Sentiment_Subjectivity    26863
dtype: int64

In [53]:
print(df_apps_raw.duplicated().sum())


483


In [54]:
print(df_apps_raw['App'].duplicated().sum())

1181


In [55]:
print(df_reviews_raw.duplicated().sum())

33616


In [56]:
df_apps_raw.describe()

,Rating
count,9367.000000
mean,4.193338
std,0.537431
min,1.000000
25%,4.000000
50%,4.300000
75%,4.500000
max,19.000000


In [57]:
df_reviews_raw.describe()

,Sentiment_Polarity,Sentiment_Subjectivity
count,37432.000000,37432.000000
mean,0.182146,0.492704
std,0.351301,0.259949
min,-1.000000,0.000000
25%,0.000000,0.357143
50%,0.150000,0.514286
75%,0.400000,0.650000
max,1.000000,1.000000


In [58]:
df_apps_raw[df_apps_raw['Rating']>5]

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
10472,Life Made WI-Fi Touchscreen Photo Frame,1.9,19.0,3.0M,"1,000+",Free,0,Everyone,NaN,"February 11, 2018",1.0.19,4.0 and up,NaN


In [59]:
df_apps_raw['Type'].value_counts()

Type
Free    10039
Paid      800
0           1
Name: count, dtype: int64

In [60]:
df_reviews_raw['Sentiment'].value_counts()

Sentiment
Positive    23998
Negative     8271
Neutral      5163
Name: count, dtype: int64

### Data Issues
- One broken row (10472): its values moved to the wrong columns
- Missing values: Rating (1,474), Current Ver (8), Android Ver (3), Type (1), Content Rating (1)
- Duplicate rows (483) and duplicate app names (1,181)
- Number columns stored as text: Reviews, Size, Installs, Price
- Date stored as text: Last Updated
- Columns we don't need: Android Ver, Current Ver
- Column names have spaces
- In reviews: 26,868 rows without text and 33,616 duplicate rows